### PR TITLE
Bump WPILib to beta 2

### PR DESCRIPTION
**Pull request by mcm001**

### CODE REVIEW

**Comment by mdurrani808** [comment]
would it be possible to just make those all one big global variable so you just need to update it in one place or does it not work like that

**Comment by srimanachanta** [comment]
Bump gradle to 8.4 while you are at it

File: photonlib-cpp-examples/apriltagExample/simgui-window.json
Comment (srimanachanta) [inline photonlib-cpp-examples/apriltagExample/simgui-window.json:1]: This file should be unchanged

File: shared/common.gradle [hunk 1]
```diff
@@ -43,6 +43,9 @@ dependencies {
 
     // test stuff
     testImplementation("org.junit.jupiter:junit-jupiter:5.8.2")
+
+    // wpilib serde
```
Comment (srimanachanta) [inline shared/common.gradle:47]: stuff?

**Review by srimanachanta — CHANGES_REQUESTED** [review]

File: shared/common.gradle (hunk 1 above)
Comment (mcm001) [reply shared/common.gradle:47]: serde. Serialization/deserialization

**Review by srimanachanta — APPROVED** [review]

**Pull request merged**